tests/php/Core/ClassInfoTest.php

```diff
@@ -266,9 +268,52 @@ public function testClassesWithExtensionWithDynamicallyRemovedExtensions()
         );
     }
 
-    /**
-     * @dataProvider provideClassSpecCases
-     */
+    /** @dataProvider provideHasMethodCases */
+    public function testHasMethod($object, $method, $output)
+    {
+        $this->assertEquals(
+            $output,
+            ClassInfo::hasMethod($object, $method)
+        );
+    }
+
+    public function provideHasMethodCases()
+    {
+        return [
+            'Basic object' => [
+                new DateTime(),
+                'format',
+                true,
+            ],
+            'CustomMethod object' => [
+                new HasMethod(),
+                'example',
+                true,
+            ],
+            'FQCN' => [
+                '\DateTime',
```

Comment:
I think an unqualified classname that lives in the root namespace like 'DateTime', 'format' would be valid, so worth adding that too

Reply:
Added 👍